juju/client/proxy/factory.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from juju.client.proxy.kubernetes.proxy import KubernetesProxy
+
+class ProxyTypeNotDefinedError(Exception):
+    pass
+
+class UnknownProxyTypeError(Exception):
+    pass
+
+class Factory(ABC):
+
+    @abstractmethod
+    def proxy_from_config(self, conf):
+        raise NotImplementedError()
+
+class JujuDataFactory(Factory):
+
+    def proxy_from_config(self, conf):
+        if conf is None:
+            return None
+
+        if "type" not in conf:
```

Comment:
Would be good to use a consistent quote style, single or double. I lean towards single as that's what's Python's `repr()` produces, and from a quick scan it looks like that's what this project prefers too. Also elsewhere in this PR.